client/src/views/Knowledge/components/Graphs/convertDataToGraferV4.ts

```diff
@@ -0,0 +1,275 @@
+import { DataSource } from '@dekkai/data-source';
+import { parseJSONL } from '@/utils/FileLoaderUtil';
+import alphaShape from 'alpha-shape';
+
+export interface LayoutInfo {
+    nodes: string;
+    nodesFile: File;
+    nodeAtts: string;
+    nodeAttsFile: File;
+    nodeLayout: string;
+    nodeLayoutFile: File;
+    groups: string;
+    groupsFile: File;
+    alpha: number;
+    level: number;
+    levelCount: number;
+    maxLabelLength: number;
+    topGroupThreshold: number;
+    pointRadius: number;
+    positionScale: number;
+}
+
+export interface GroupColor {
+    id: string;
+    level: number;
+    primary: number;
+    inherited: Set<number> | number[];
+    top: boolean;
+}
+
+export interface GroupCentroid {
+    id: string;
+    point: string;
+    label: string;
+    color: number;
+    level: number;
+    top: boolean;
+}
+
+export interface GroupHullEdge {
+    id: string;
+    source: string;
+    target: string;
+    sourceColor: number;
+    targetColor: number;
+    group: string;
+    level: number;
+}
+
+// export interface KnowledgeNodeData extends BasicNodeData {
+//     level: number;
+// }
+
+export interface GraferData {
+    points;
+    nodes: any[];
+    shapes: GroupHullEdge[];
+    colors: GroupColor[];
+    centroids: GroupCentroid[];
+}
+
+export async function convertDataToGraferV4 (info: LayoutInfo): Promise<GraferData> {
+  let lineNumber;
+
+  // variable to hold the number of colors used
+  let colorIndex = 0;
+
+  // load the points
+  console.log('Loading points...');
```

Comment:
These console log's should remain for debugging purposes until we've transferred this over to an offline infrastructure. (ie. instead of computing clusters in the browser; pre-compute and store files in S3)